src/services/TodoService.ts

```diff
@@ -0,0 +1,158 @@
+import { useEffect } from 'react';
+import { addTodos, delateTodos, patchTodos } from '../api/todos';
+import { useTodoContext } from '../utils/hooks/useTodoContext';
+import { ErrorMessages } from '../types/ErrorMessages/ErrorMessages';
+import { Todo } from '../types/Todo/Todo';
+
+export const TodoService = () => {
```

Comment:
you've made something similar to a component to be able to call hooks inside. But it's not a component and such an approach is wrong. Instead, make custom hook from it

```suggestion
export const useTodoActions = () => {
```